test/AutoDiff/Parse/transpose_attr_parse.swift

```diff
@@ -78,16 +78,28 @@ func transpose(v: Float) -> Float
 @transpose(of: foo, wrt: (0, v))
 func transpose(v: Float) -> Float
 
-// expected-error @+2 {{expected ')' in 'transpose' attribute}}
+// NOTE: The "expected ',' separator" diagnostic is not ideal.
```

Comment:
Note: as explained in the comment, the new diagnostic isn't great here.
The old diagnostic wasn't any better though, so I wouldn't say it's a regression.

```swift
@transpose(of: Swift.Float.+(_:_))
func transpose(v: Float) -> Float
```
Before:
```console
transpose.swift:1:29: error: expected ')' in 'transpose' attribute
@transpose(of: Swift.Float.+(_:_))
                            ^
transpose.swift:1:29: error: expected declaration
@transpose(of: Swift.Float.+(_:_))
                            ^
```
After:
```console
transpose.swift:1:29: error: expected ',' separator
@transpose(of: Swift.Float.+(_:_))
                            ^
transpose.swift:1:29: error: expected declaration
@transpose(of: Swift.Float.+(_:_))
                            ^
```

It's not obvious how to improve the diagnostic, so I'll defer it for now.